### ISSUE TITLE
Async functions without yields which return async iterables have wrong signature

### ISSUE THREAD

**Issue by audoh-tickitto** [microsoft/pylance-release#1140]
## Environment data

-   Language Server version: 2021.4.0 (pyright 2918da1d)
-   OS and version: Ubuntu 20.04.2 LTS
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.2 acquired via pyenv 1.2.23

## Expected behaviour

Non-generator (no yield) async functions which return async iterables, generators etc should have a return type of `Coroutine[Any, Any, AsyncIterable]` _(replace AsyncIterable with AsyncGenerator or other relevant type)_

## Actual behaviour

Non-generator (no yield) async functions which return async iterables, generators etc have a return type of `AsyncIterable` _(replace AsyncIterable with AsyncGenerator or other relevant type)_

## Code Snippet / Additional information

```python
import asyncio
from typing import AsyncGenerator, AsyncIterable, List


async def get_data() -> List[int]:
    await asyncio.sleep(1)
    return [1, 2, 3]


async def generate(nums: List[int]) -> AsyncGenerator[str, None]:
    for n in nums:
        await asyncio.sleep(1)
        yield f"The number is {n}"


async def get_generator() -> AsyncGenerator[str, None]:
    data = await get_data()
    return generate(data)


async def demo_bug() -> None:
    gen = await get_generator()  # Pylance: "AsyncGenerator[str, None]" is not awaitable
    async for s in gen:
        print(s)


loop = asyncio.get_event_loop()
loop.run_until_complete(demo_bug())
```

When run in Python, the expected output is produced:

```
The number is 1
The number is 2
The number is 3
```

**Comment by erictraut** [comment]
Thanks for the bug report. Pylance wasn't correctly handling the case where an async method had a declared return type of `AsyncGenerator` but was not itself a generator (i.e. had no `yield` statement). In that case, the effective return type of the method needs to be wrapped in a `Coroutine` type.

**Comment by erictraut** [comment]
This will be fixed in the next release of Pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.4.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202141-14-april-2021
